### PR TITLE
Azure FILE-SYSTEM-VERIFICATION-TESTS has SMB3.0 supporting distro list

### PR DESCRIPTION
**Pull request by juhlee-microsoft**
Updated the comment with shortened url of Azure filesystem guideline for Linux
Also added the kernel version filter inside the test script.

[LISAv2 Test Results Summary]
Test Run On           : 09/23/2020 18:01:54
ARM Image Under Test  : Canonical : UbuntuServer : 16.04-LTS : latest
Test Priority         : 0,1,2,3
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:1:59

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 STORAGE              FILE-SYSTEM-VERIFICATION-TESTS-GENERIC                                            PASS               115.62 

      ARMImageName: Canonical UbuntuServer 16.04-LTS latest, DiskType: Managed, TestLocation: westus2, Kernel Version: 4.15.0-1096-azure

[LISAv2 Test Results Summary]
Test Run On           : 09/23/2020 18:02:03
ARM Image Under Test  : Canonical : UbuntuServer : 18.04-LTS : latest
Test Priority         : 0,1,2,3
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:1:40

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 STORAGE              FILE-SYSTEM-VERIFICATION-TESTS-GENERIC                                            PASS                96.17 
      ARMImageName: Canonical UbuntuServer 18.04-LTS latest, DiskType: Managed, TestLocation: westus2, Kernel Version: 5.4.0-1026-azure

[LISAv2 Test Results Summary]
Test Run On           : 09/23/2020 18:02:24
ARM Image Under Test  : OpenLogic : CentOS : 7.4 : latest
Test Priority         : 0,1,2,3
Total Test Cases      : 1 (0 Passed, 0 Failed, 0 Aborted, 1 Skipped)
Total Time (dd:hh:mm) : 0:0:4

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 STORAGE              FILE-SYSTEM-VERIFICATION-TESTS-GENERIC                                         **SKIPPED**                  1.8 
      ARMImageName: OpenLogic CentOS 7.4 latest, DiskType: Managed, TestLocation: westus2, Kernel Version: 3.10.0-693.21.1.el7.x86_64

[LISAv2 Test Results Summary]
Test Run On           : 09/23/2020 18:02:13
ARM Image Under Test  : RedHat : RHEL : 7.4 : latest
Test Priority         : 0,1,2,3
Total Test Cases      : 1 (0 Passed, 0 Failed, 0 Aborted, 1 Skipped)
Total Time (dd:hh:mm) : 0:0:4

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 STORAGE              FILE-SYSTEM-VERIFICATION-TESTS-GENERIC                                         SKIPPED                 1.79 
      ARMImageName: RedHat RHEL 7.4 latest, DiskType: Managed, TestLocation: westus2, Kernel Version: 3.10.0-693.58.1.el7.x86_64

[LISAv2 Test Results Summary]
Test Run On           : 09/23/2020 18:02:10
ARM Image Under Test  : SUSE : sles-15-sp2 : gen1 : latest
Test Priority         : 0,1,2,3
Total Test Cases      : 1 (1 Passed, 0 Failed, 0 Aborted, 0 Skipped)
Total Time (dd:hh:mm) : 0:1:21

   ID TestArea             TestCaseName                                                                TestResult TestDuration(in minutes) 
-------------------------------------------------------------------------------------------------------------------------------------------
    1 STORAGE              FILE-SYSTEM-VERIFICATION-TESTS-GENERIC                                            PASS                75.95 
      ARMImageName: SUSE sles-15-sp2 gen1 latest, DiskType: Managed, TestLocation: westus2, Kernel Version: 5.3.18-18.18-azure

### CODE REVIEW

**Review by lpereira — DISMISSED** [review]
LGTM

File: Testscripts/Linux/xfstesting.sh [hunk 1]
```diff
@@ -169,6 +169,15 @@ ConfigureCIFS() {
 }
 
 Main() {
+    # Get VM kernel version. Recommend to 3.12 or later for SMB3.0 in Azure.
+    mj=$(uname -r | cut -d '-'  -f1 | cut -d '.' -f 1)
+    mn=$(uname -r | cut -d '-'  -f1 | cut -d '.' -f 2)
+    if [ $mj -lt 4 ] && [ $mn -lt 12 ]; then
```
Comment (johnsongeorge-w) [inline Testscripts/Linux/xfstesting.sh]: Here we cannot use kernel version directly. Distro providers (like RHEL) backports the patches to old kernel.  Add first level check for distro and then kernel.

As per the document it is supported from RH7.5+, so the base kernel version would be ```3.10.0-862```

**Review by johnsongeorge-w — CHANGES_REQUESTED** [review]

File: Testscripts/Linux/xfstesting.sh (hunk 1 above)
Comment (juhlee-microsoft) [reply Testscripts/Linux/xfstesting.sh:175]: If distro is RHEL or CentOS, then we set the baseline to 3.10.0. OK?

Comment (juhlee-microsoft) [reply Testscripts/Linux/xfstesting.sh]: What is "the document"? Where do you find this kernel version? 3.10.0-862

Comment (johnsongeorge-w) [reply Testscripts/Linux/xfstesting.sh]: https://access.redhat.com/articles/3078

Comment (juhlee-microsoft) [reply Testscripts/Linux/xfstesting.sh]: This has RHEL 7.5 errata date with this kernel version. But I am still confused about your comment.
Why do you think we should support RHEL 7.5 and this kernel? I reviewed the contents in 2018-04-10 RHEA-2018:0700 but could not find any clue for SMB3.0 bug fix/change.
I still need your clarification why you mentioned "As per the document it is supported from RH7.5+". @johnsongeorge-w 
I can not find patch/bug fix information yet.

Microsoft document is not clearly defined the distro version. "7.5+" is ambiguious. Also, SMB 3.0 encryption is added in 4.11

Comment (juhlee-microsoft) [reply Testscripts/Linux/xfstesting.sh]: We looked into the database result, and CentOS/RHEL .7.6 or later version passed this test. From the regression perspective, I put the filter for rhel/centos 7.6+. Fixed in d2a2f0a70002124d6a6e163a4ee51cc39fd364b9

**Review by johnsongeorge-w — APPROVED** [review]

**Review by abhimarathe — APPROVED** [review]

**Comment by lpereira** [comment]
Looked here at a machine running CentOS, and a more future-proof way of whitelisting this is looking at the "cifs" module, and checking if it's 2.02 or later (according to the Samba Wiki).  The version can be obtained with `modinfo cifs | grep ^version:`.

**Comment by johnsongeorge-w** [comment]
@juhlee-microsoft  This change will cause all filesystem test using this script to skip (<RH7.6).

**Pull request merged**